### PR TITLE
feat: upgrade the gitlab agent functions to configure the proper gitlab agent in CI/CD

### PR DESCRIPTION
**Pull request by Monska85**

### CODE REVIEW

File: scripts/src/functions.bash [hunk 1]
```diff
@@ -114,16 +120,98 @@ create-ns-and-developer-role-bindings() {
...
+    fi
+
+    _setup-gitlab-agent-kubernetes-context "${DEVELOP_GITLAB_AGENT_PROJECT}" "${DEVELOP_GITLAB_AGENT_ID}"
+    return
+  fi
+
+  # If the current branch is a production ones and the production variables are set, use the production GitLab Agent.
+  if echo "${CI_COMMIT_REF_SLUG}" | grep -qE "^(${NON_DEVELOP_BRANCHES_REGEX})$"; then
+    if [ -n "${PRODUCTION_GITLAB_AGENT_PROJECT:-}" ] && [ -n "${PRODUCTION_GITLAB_AGENT_ID:-}" ]; then
+      reutrn
```
Comment (paolomainardi) [inline scripts/src/functions.bash:211]: typo

Comment (Monska85) [reply scripts/src/functions.bash]: Fixed

File: scripts/src/functions.bash [hunk 2]
```diff
@@ -114,16 +120,98 @@ create-ns-and-developer-role-bindings() {
...
+  fi
+
+  # If the current branch is a production ones and the production variables are set, use the production GitLab Agent.
+  if echo "${CI_COMMIT_REF_SLUG}" | grep -qE "^(${NON_DEVELOP_BRANCHES_REGEX})$"; then
+    if [ -n "${PRODUCTION_GITLAB_AGENT_PROJECT:-}" ] && [ -n "${PRODUCTION_GITLAB_AGENT_ID:-}" ]; then
+      return
+    fi
+
+    _setup-gitlab-agent-kubernetes-context "${PRODUCTION_GITLAB_AGENT_PROJECT}" "${PRODUCTION_GITLAB_AGENT_ID}"
+    return
```
Comment (paolomainardi) [inline scripts/src/functions.bash:215]: I would add a new function to print that the agent is not configured.

```bash

_gitlab_agent_exit_not_configured() {
    echo "Gitlab agent not configured"
    // pseudocode to print a custom message if present.
    if arg(1)
      echo ": arg(1)"
    return 
}
```

Comment (Monska85) [reply scripts/src/functions.bash:217]: > I would add a new function to print that the agent is not configured.
> 
> ```shell
> _gitlab_agent_exit_not_configured() {
>     echo "Gitlab agent not configured"
>     // pseudocode to print a custom message if present.
>     if arg(1)
>       echo ": arg(1)"
>     return 
> }
> ```

The message for non-configured-agent is printed here: https://github.com/sparkfabrik/spark-k8s-deployer/pull/174/files#diff-96524718f63a467ac7a74e21082c918aa6e57208e1d794964a8c88fbb65f193cR159

Is this not enough?

Comment (Monska85) [reply scripts/src/functions.bash:217]: > Altra cosa che farei è un end banner, così da avere le sezioni ben separate.

Usata la stessa funzione `print-banner` con l'end della sezione.

**Pull request merged**